### PR TITLE
Fix a bunch of little issues with the new auth API

### PR DESCRIPTION
**Pull request by samhorlbeck**

### CODE REVIEW

File: docs-exp/auth-types.actioncodeinfo.data.md [hunk 1]
```diff
@@ -4,13 +4,13 @@
...
+For the  action, which allows a user to verify the email before updating it, this object contains a `previousEmail` field with the user account's email address before updating. After the action completes, the user's email address will be updated to the value in the `email` field from the value in `previousEmail` field.
 
-For the [Operation.REVERT\_SECOND\_FACTOR\_ADDITION](./auth-types.operation.revert_second_factor_addition.md) action, which allows a user to unenroll a newly added second factor, this object contains a `multiFactorInfo` field with the information about the second factor. For phone second factor, the `multiFactorInfo` is a [MultiFactorInfo](./auth-types.multifactorinfo.md) object, which contains the phone number.
+For the  action, which allows a user to unenroll a newly added second factor, this object contains a `multiFactorInfo` field with the information about the second factor. For phone second factor, the `multiFactorInfo` is a [MultiFactorInfo](./auth-types.multifactorinfo.md) object, which contains the phone number.
```
Comment (avolkovi) [inline docs-exp/auth-types.actioncodeinfo.data.md:13]: looks like a broken @link in the docs

**Review by avolkovi — DISMISSED** [review]
Can you also pull in the changes from https://github.com/firebase/firebase-js-sdk/pull/3990 so I can close that PR

File: packages-exp/auth-exp/src/core/providers/oauth.ts [hunk 2]
```diff
@@ -24,7 +24,7 @@ import { OAuthCredential } from '../credentials/oauth';
 
 export type CustomParameters = Record<string, string>;
 
-interface CredentialParameters {
+export interface OAuthCredentialOptions {
```
Comment (avolkovi) [inline packages-exp/auth-exp/src/core/providers/oauth.ts:27]: should this live in auth-types-exp?

File: packages-exp/auth-exp/src/core/user/additional_user_info.test.ts [hunk 3]
```diff
@@ -35,13 +31,13 @@ import { User, UserCredential } from '../../model/user';
 import { testAuth, testUser } from '../../../test/helpers/mock_auth';
 
 describe('core/user/additional_user_info', () => {
-  const userProfileWithLogin: UserProfile = {
+  const userProfileWithLogin: Record<string, unknown> = {
```
Comment (avolkovi) [inline packages-exp/auth-exp/src/core/user/additional_user_info.test.ts:34]: is this type necessary here?

Comment (samhorlbeck) [reply packages-exp/auth-exp/src/core/user/additional_user_info.test.ts:35]: Yes; object literals should have a strong type specified

File: packages-exp/auth-exp/src/core/providers/oauth.ts (hunk 2 above)
Comment (samhorlbeck) [reply packages-exp/auth-exp/src/core/providers/oauth.ts]: no; this is an internal-only interface

**Review by hsubox76 — APPROVED** [review]

**Pull request merged**